### ISSUE TITLE
XKit.interface.get_posts returns the post being edited in the post window

### ISSUE THREAD

**Issue by finagle29** [new-xkit/XKit#676]
This causes things like this to happen
![screen shot 2015-10-25 at 7 21 10 pm](https://cloud.githubusercontent.com/assets/5816532/10720205/d5f28a9e-7b6a-11e5-8ec0-01b935a7de57.png)
when RDO is enabled and ER is not (test post http://120983019283-testing.tumblr.com/post/131826469124/teen-stanley-b3-and-stanford-c3-hugging-if).

**Comment by 0xazure** [comment]
I'm not sure I understand the bug, or the screenshot.  `XKit.interface.get_posts()` returns all posts not containing a given CSS class.  What is the behaviour you're expecting?

**Comment by nightpool** [comment]
@0xazure it shouldn't return the post edit window. there's a different
function for that.

edit windows aren't posts in the way people using post_listener are
expecting

**Comment by 0xazure** [comment]
> there's a different function for that.

You'll have to direct me to the code, because a perusal of `XKit.interface` in both `xkit_patches` and the docs hasn't turned anything up.

As far as filtering out the post window "post", we might be able to exclude `.post-form`?  Though that class is higher up the DOM tree; there's nothing at the same level as `.post` that's unique to the post window as far as I can see.  My jQ is rusty though, so I'm not sure if that would work.

**Comment by nightpool** [comment]
@0xazure post_window.listener

**Comment by 0xazure** [comment]
Oh, the listener mechanism.  Sorry, I was only thinking of functions like `get_posts()` that you'd have to call manually.

**Comment by nightpool** [comment]
yes, the edit window is not a post on the dashboard and therefore shouldn't
be returned by get_posts.

**Comment by nightpool** [comment]
Not sure I still find this compelling, but we should make sure this isn't an RDO bug
